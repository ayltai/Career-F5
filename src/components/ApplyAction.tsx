import { RocketLaunch as RocketLaunchIcon, } from '@mui/icons-material';
import { Button, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const ApplyAction = ({
    onClick,
} : {
    onClick : () => void,
}) => {
    const { t, } = useTranslation();

    return (
        <Button
            color='primary'
            variant='contained'
            startIcon={
                <RocketLaunchIcon sx={{
                    color : 'white',
                }} />
            }
            onClick={onClick}>
            <Typography
                color='white'
                variant='button'>
                {t('action_apply')}
            </Typography>
        </Button>
    );
};
