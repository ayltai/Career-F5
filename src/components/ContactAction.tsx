import { WhatsApp as WhatsAppIcon, } from '@mui/icons-material';
import { Button, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

const handleClick = () => {
    if (process.env.REACT_APP_WHATSAPP_PHONE_NUMBER) {
        const newWindow = window.open(`https://wa.me/${process.env.REACT_APP_WHATSAPP_PHONE_NUMBER}`, '_blank', 'noopener,noreferrer');
        if (newWindow) newWindow.opener = null;
    }
};

export const ContactAction = () => {
    const { t, } = useTranslation();

    return (
        <Button
            variant='contained'
            color='secondary'
            startIcon={
                <WhatsAppIcon sx={{
                    color : 'white',
                }} />
            }
            onClick={handleClick}>
            <Typography
                color='white'
                variant='button'>
                {t('whatsapp')}
            </Typography>
        </Button>
    );
};
