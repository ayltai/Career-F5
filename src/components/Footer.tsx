import { Box, Container, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const Footer = () => {
    const { t, } = useTranslation();

    return (
        <Box
            sx={{
                backgroundColor: 'black',
            }}
            width='100%'>
            <Container maxWidth='lg'>
                <Box
                    paddingY={4}
                    textAlign='center'>
                    <Typography
                        color='white'
                        variant='overline'>
                        {t('copyright')}
                    </Typography>
                </Box>
            </Container>
        </Box>
    );
};
