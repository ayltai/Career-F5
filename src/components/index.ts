export { ApplyAction, } from './ApplyAction';
export { ApplyDialog, } from './ApplyDialog';
export { ContactAction, } from './ContactAction';
export { Footer, } from './Footer';
export { PaymentPlan, } from './PaymentPlan';
export { RangeIndicator, } from './RangeIndicator';
export { TopBar, } from './TopBar';
