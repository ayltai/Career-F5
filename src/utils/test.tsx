import { ThemeProvider, } from '@mui/material';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { preferencesReducer, } from '../states';
import { AppTheme, } from '../styles';

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences : preferencesReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?  : PreloadedState<RootState>,
    store?           : Store<RootState>,
    [ rest : string] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme('en')}>
                {children}
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
