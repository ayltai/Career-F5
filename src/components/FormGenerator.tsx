import { Button, Card, CardActions, CardContent, Typography, useMediaQuery, } from '@mui/material';
import { GridContainer, widgets, } from '@ui-schema/ds-material';
import { createEmptyStore, createOrderedMap, injectPluginStack, isInvalid, relT, StoreSchemaType, storeUpdater, TranslatorContext, UIMetaProvider, UIStoreProvider, } from '@ui-schema/ui-schema';
import React, { useCallback, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { handleError, } from '../utils';

const GridStack = injectPluginStack(GridContainer);

export type FormGeneratorProps = {
    title?       : string,
    description? : string,
    definition   : object,
    onSubmit     : (data : Record<string, string>) => Promise<boolean>,
};

export const FormGenerator = ({
    elevated = true,
    title,
    description,
    definition,
    onSubmit,
} : FormGeneratorProps & {
    elevated? : boolean,
}) => {
    const storeSchema = createOrderedMap({
        type : 'object',
        ...definition,
    });

    const [ store,        setStore,        ] = useState(() => createEmptyStore(storeSchema.get('type')));
    const [ showValidity, setShowValidity, ] = useState(false);
    const [ isSubmitting, setIsSubmitting, ] = useState(false);

    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    const handleChange = useCallback((actions : any) => setStore(storeUpdater(actions)), [ setStore,]);

    const handleSubmit = () => {
        setShowValidity(true);

        if (!isInvalid(store.getValidity())) {
            setIsSubmitting(true);

            onSubmit(store.valuesToJS() as Record<string, string>).then(success => {
                if (success) {
                    setStore(createEmptyStore(storeSchema.get('type')));

                    setShowValidity(false);
                }
            }).then(() => setIsSubmitting(false)).catch(handleError);
        }
    };

    const translate = (text : string, context? : TranslatorContext, schema? : StoreSchemaType) => {
        const schemaT = relT(schema, context);

        return schemaT ? schemaT : t(text);
    };

    return (
        <Card
            sx={{
                borderRadius : 4,
                padding      : desktopMode ? 4 : 1,
            }}
            elevation={elevated ? 4 : 0}>
            <CardContent>
                {title && (
                    <Typography
                        gutterBottom
                        variant='h5'
                        fontWeight='bold'>
                        {t(title)}
                    </Typography>
                )}
                {description && (
                    <Typography
                        gutterBottom
                        variant='body1'>
                        {t(description)}
                    </Typography>
                )}
                <UIMetaProvider
                    t={translate}
                    widgets={widgets}>
                    <UIStoreProvider
                        showValidity={showValidity}
                        store={store}
                        onChange={handleChange}>
                        <GridStack
                            isRoot
                            schema={storeSchema} />
                    </UIStoreProvider>
                </UIMetaProvider>
            </CardContent>
            <CardActions>
                <Button
                    sx={{
                        marginX : 1,
                    }}
                    disabled={isSubmitting || !!isInvalid(store.getValidity())}
                    variant='contained'
                    onClick={handleSubmit}>
                    <Typography
                        color='white'
                        variant='button'>
                        {t(isSubmitting ? 'action_submitting' : 'action_submit')}
                    </Typography>
                </Button>
            </CardActions>
        </Card>
    );
};
