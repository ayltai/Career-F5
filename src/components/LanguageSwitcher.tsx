import { Language as LanguageIcon, } from '@mui/icons-material';
import { Box, IconButton, Menu, MenuItem, useMediaQuery, } from '@mui/material';
import React, { MouseEvent, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { useAppDispatch, useAppSelector, } from '../hooks';
import { setLanguage, } from '../states';
import { handleError, } from '../utils';

const LanguageMenuItem = ({
    selected,
    name,
    onLanguageChange,
} : {
    selected?         : boolean,
    name?             : string,
    onLanguageChange? : () => void,
}) => (
    <MenuItem
        selected={selected}
        onClick={onLanguageChange}>
        {name}
    </MenuItem>
);

export const LanguageSwitcher = () => {
    const [ anchorElement, setAnchorElement, ] = useState<null | HTMLElement>(null);

    const language    = useAppSelector(state => state.preferences.language);
    const dispatch    = useAppDispatch();
    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, i18n, } = useTranslation();

    const handleClick = (event : MouseEvent<HTMLButtonElement>) => setAnchorElement(event.currentTarget);

    const handleClose = () => setAnchorElement(null);

    const handleLanguageChangeEn = () => {
        dispatch(setLanguage('en'));

        i18n.changeLanguage('en').catch(handleError);

        setAnchorElement(null);
    };

    const handleLanguageChangeZh = () => {
        dispatch(setLanguage('zh'));

        i18n.changeLanguage('zh').catch(handleError);

        setAnchorElement(null);
    };

    return (
        <Box marginLeft={desktopMode ? 1 : 'auto'}>
            <IconButton
                size='large'
                edge='end'
                color='inherit'
                onClick={handleClick}>
                <LanguageIcon />
            </IconButton>
            <Menu
                anchorEl={anchorElement}
                open={Boolean(anchorElement)}
                onClose={handleClose}>
                <LanguageMenuItem
                    selected={language === 'en'}
                    name={t<string>('language_en')}
                    onLanguageChange={handleLanguageChangeEn} />
                <LanguageMenuItem
                    selected={language === 'zh'}
                    name={t<string>('language_zh')}
                    onLanguageChange={handleLanguageChangeZh} />
            </Menu>
        </Box>
    );
};
