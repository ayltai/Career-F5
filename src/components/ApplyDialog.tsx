import { Close as CloseIcon, } from '@mui/icons-material';
import { Box, Dialog, DialogContent, DialogContentText, DialogTitle, IconButton, Typography, useMediaQuery, useTheme, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

import { FormGenerator, FormGeneratorProps, } from './FormGenerator';

export const ApplyDialog = ({
    open = false,
    onClose,
    title,
    description,
    definition,
    onSubmit,
} : FormGeneratorProps & {
    open?    : boolean,
    onClose? : () => void,
}) => {
    const theme = useTheme();

    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    return (
        <Dialog
            open={open}
            onClose={onClose}>
            <DialogTitle>
                <Box
                    paddingX={desktopMode ? 4 : 1}
                    paddingTop={desktopMode ? 4 : 1}>
                    <Typography variant='h5'>
                        {title}
                    </Typography>
                    <IconButton
                        sx={{
                            position : 'absolute',
                            right    : desktopMode ? 16 : 8,
                            top      : desktopMode ? 16 : 8,
                            color    : theme.palette.text.secondary,
                        }}
                        onClick={onClose}>
                        <CloseIcon />
                    </IconButton>
                </Box>
            </DialogTitle>
            <DialogContent>
                <DialogContentText>
                    <Typography
                        gutterBottom
                        paddingX={desktopMode ? 8 : 4}
                        color='text.secondary'
                        variant='subtitle1'
                        fontStyle='italic'>
                        {t('apply_note')}
                    </Typography>
                    <Typography
                        gutterBottom
                        paddingX={desktopMode ? 4 : 1}
                        paddingTop={2}
                        color='text.primary'
                        variant='body1'>
                        {description}
                    </Typography>
                </DialogContentText>
                <FormGenerator
                    elevated={false}
                    definition={definition}
                    onSubmit={onSubmit} />
            </DialogContent>
        </Dialog>
    );
};
