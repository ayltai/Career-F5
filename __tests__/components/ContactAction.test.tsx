import React from 'react';

import { fireEvent, render, } from '../../src/utils/test';
import { ContactAction, } from '../../src/components/ContactAction';

describe('ContactAction', () => {
    it('renders correctly', () => {
        expect(render(<ContactAction />)).toMatchSnapshot();
    });

    it('responses to click', () => {
        process.env.REACT_APP_WHATSAPP_PHONE_NUMBER = '5511999999999';

        window.open = jest.fn();

        const { getByRole, } = render(<ContactAction />);

        fireEvent.click(getByRole('button'));

        expect(window.open).toBeCalledTimes(1);
        expect(window.open).toBeCalledWith(`https://wa.me/${process.env.REACT_APP_WHATSAPP_PHONE_NUMBER}`, '_blank', 'noopener,noreferrer');
    });
});
