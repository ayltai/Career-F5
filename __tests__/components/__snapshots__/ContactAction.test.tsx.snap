// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactAction renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-6r1083-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
            data-testid="WhatsAppIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-button css-1we6fc-MuiTypography-root"
        >
          whatsapp
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-6r1083-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
          data-testid="WhatsAppIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
          />
        </svg>
      </span>
      <span
        class="MuiTypography-root MuiTypography-button css-1we6fc-MuiTypography-root"
      >
        whatsapp
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
